### PR TITLE
add error banner for failed migration cutover and cancel migration

### PR DESCRIPTION
**Pull request by brian-harris**
This PR fixes
* Add custom eslint rule to require void or await before all async calls
* Fix BackupFileInfo canceled status field name
* Add error handling and error message banner to cutover and cancel migration failures 
  - VSO Task 1353075: PPG - ADS UI , surface cutover operation failed message

Cutover error
image

Cancel migration error
image

### CODE REVIEW

File: extensions/sql-migration/src/api/utils.ts [hunk 1]
```diff
@@ -230,6 +230,14 @@ export function get12HourTime(date: Date | undefined): string {
 	return (date ? date : new Date()).toLocaleTimeString([], localeTimeStringOptions);
 }
 
+export function displayDialogErrorMessage(dialog: window.Dialog, text: string, error: Error): void {
+	dialog.message = {
```
Comment (ysy68251435) [inline extensions/sql-migration/src/api/utils.ts:234]: dialog message doesn't need to handle localization?

Comment (ysy68251435) [reply extensions/sql-migration/src/api/utils.ts:234]: I noticed other places use Localize('','') but here not.

**Review by ysy68251435 — APPROVED** [review]
Looks good to me(siyao)

File: extensions/sql-migration/src/api/utils.ts (hunk 1 above)
Comment (brian-harris) [reply extensions/sql-migration/src/api/utils.ts:234]: this is a function that is taking parameters that use localized strings.

**Pull request merged**